ete4/smartview/ete/gardening.pyx

```diff
@@ -52,18 +40,17 @@ def add_intermediate(node):
     "Add an intermediate parent to the given node and return it"
     parent = node.up
 
-    pos = parent.children.index(node)  # position of node in parent's children
-
-    parent.children.pop(pos)  # detach from parent
+    parent.children.remove(node)  # detach from parent
 
     intermediate = Tree('')  # create intermediate node
     intermediate.children = [node]
+    node.up = intermediate
```

Comment:
Uhm, I see that the setter in `tree.pyx:_set_children` already does that. So this line is indeed not necessary.

I missed that initially. Let me "fix" it.

Reply:
I changed the setter to have that behavior. I think it makes sense for a node to be its children's parent, at least for nodes...

Reply:
Plus, there some other places in the code where this relationship was made explicitly and I believe it should be intrinsic to the Tree class... Correct me if I am wrong. 
I missed this one and did not remove it 🙃

Reply:
I agree, it seems correct that when you make a node the child of another, it will automatically set the node's parent too.

By the way, I don't think it happens with `node.children.append(...)` and I find it a bit confusing (I guess one should use `node.add_child(...)' instead!).